src/pf.lua

```diff
@@ -24,13 +24,20 @@ function compile_filter(filter_str, opts)
          return libpcap.offline_filter(bytecode, header, P) ~= 0
       end
    elseif opts.bpf then
-      local bytecode = libpcap.compile(filter_str, dlt)
+      local bytecode
+      if opts.noopt then
+         bytecode = libpcap.compile(filter_str, dlt, true)
```

Comment:
I understand why you made the option negative, because the default "opts.foo" is false if foo is not there.  Still, i think it's probably more readable if we avoid negative options, and especially we should avoid them propagating farther (as to arguments, as here).

Can we therefore add some defaults to the opts table?  E.g. for all keys in the opts table, verify they are in the defaults table (i.e. defaults[key] ~= nil); and then for all keys in the defaults table, if they are not in the opts table, then we add them with the defaults[key] value.  We can make a helper that returns a fresh checked opts dict, without mutating the incoming opts dict.  Then we can make the option "optimize" instead of "noopt", and likewise for the argument to libpcap.compile.  WDYT?